### PR TITLE
Misc fixes over the last day or two

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by akhilveeraghanta** [comment]
wait, still a draft, will re-request in a bit

File: src/software/ai/hl/stp/play/free_kick_play.cpp [hunk 1]
```diff
@@ -22,18 +22,15 @@ FreeKickPlay::FreeKickPlay(std::shared_ptr<const PlayConfig> config)
...
-    // Make sure we don't interfere with the cornerkick play
-    return world.gameState().isOurFreeKick() &&
-           min_dist_to_corner >= CornerKickPlay::BALL_IN_CORNER_RADIUS;
+    return world.gameState().isOurFreeKick();
 }
 
 bool FreeKickPlay::invariantHolds(const World &world) const
 {
-    return world.gameState().isOurFreeKick();
+    LOG(DEBUG) << world.getTeamWithPossessionConfidence() << " : " << world.getTeamWithPossession();
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/free_kick_play.cpp:30]: is this still needed?

File: src/software/ai/hl/stp/play/free_kick_play.cpp [hunk 2]
```diff
@@ -22,18 +22,15 @@ FreeKickPlay::FreeKickPlay(std::shared_ptr<const PlayConfig> config)
...
+
+    return !world.gameState().isStopped() && !world.gameState().isHalted()
+        && !(world.getTeamWithPossession() == TeamSide::ENEMY && world.getTeamWithPossessionConfidence() > 1.0);
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/free_kick_play.cpp:33]: does this ever transition to shootorpass?

File: src/software/ai/hl/stp/tactic/penalty_kick/penalty_kick_fsm.h [hunk 3]
```diff
@@ -221,20 +221,21 @@ struct PenaltyKickFSM
...
+            std::optional<Robot> enemy_goalie = event.common.world.enemyTeam().goalie();
+            const Point next_shot_position =
+                evaluateNextShotPosition(enemy_goalie, event.common.world.field());
+            Point final_position = event.common.world.ball().position();
+            shot_angle           = (next_shot_position - final_position).orientation();
+            DribbleFSM::ControlParams control_params{
+                .dribble_destination       = std::optional<Point>(final_position),
+                .final_dribble_orientation = std::optional<Angle>(shot_angle),
+                // we minimize the chance of awful autokicks just as we're about to shoot
+                .allow_excessive_dribbling = true};
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/penalty_kick/penalty_kick_fsm.h:236]: why allow excessive dribbling?

**Pull request merged**